res/controllers/Reloop-Mixage.scripts.js

```diff
@@ -0,0 +1,812 @@
+    Mixage.vuMeterConnection.forEach(function(connection) { connection.disconnect(); });
+    Mixage.loopConnection.forEach(function(connection) { connection.disconnect(); });
+    Mixage.fxSelectConnection.forEach(function(connection) { connection.disconnect(); });
+    Mixage.fxOnConnection.forEach(function(connection) { connection.disconnect(); });
+
+    // Disconnect all controls from functions
+    Mixage.channels.forEach(function(channel) { Mixage.connectControlsToFunctions(channel, true); });
+
+    // all button LEDs off
+    for (let i = 0; i < 255; i++) {
+        midi.sendShortMsg(0x90, i, 0);
+    }
+};
+
+// Toggle the LED on the MIDI controller by sending a MIDI message
+Mixage.toggleLED = function(value, group, control) {
+    midi.sendShortMsg(0x90, Mixage.ledMap[group][control], value ? 0x7F : 0);
+};
+
+// Toggles the FX On LED / Off when no effect unit is activated for a channel / On when any effect unit is active for a channel
+Mixage.toggleFxLED = function(group) {
+    const fxChannel = `group_${group}_enable`;
+    const enabledFxGroups = [];
+
+    for (let i = 1; i <= Mixage.numEffectUnits; i++) {
+        enabledFxGroups.push(engine.getValue(`[EffectRack1_EffectUnit${i}]`, fxChannel));
+    }
+
+    if (enabledFxGroups.indexOf(1) !== -1) {
+        Mixage.toggleLED(ON, group, "fx_on");
+    } else {
+        Mixage.toggleLED(OFF, group, "fx_on");
+    }
+};
+
+// Flash the Reloop LED if a loop is set but currently not active
+Mixage.toggleReloopLED = function(group) {
+    if (engine.getValue(group, "loop_enabled") === 0) {
+        Mixage.toggleLED(OFF, group, "loop");
+        Mixage.toggleLED(OFF, group, "reloop");
+
+        if (engine.getValue(group, "loop_start_position") !== -1 && engine.getValue(group, "loop_end_position") !== -1) {
+            Mixage.blinkLED(Mixage.ledMap[group].reloop, group, 1000);
+        } else {
+            Mixage.blinkLED(Mixage.ledMap[group].reloop, group, 0);
+        }
+    } else {
+        Mixage.blinkLED(Mixage.ledMap[group].reloop, group, 0);
+        Mixage.toggleLoopLED(group);
+    }
+};
+
+// Turns the loop in and loop LEDs on if a loop end or start position is found, otherwise turn them off
+Mixage.toggleLoopLED = function(group) {
+    if (engine.getValue(group, "loop_start_position") !== -1) {
+        Mixage.toggleLED(ON, group, "loop");
+    } else {
+        Mixage.toggleLED(OFF, group, "loop");
+    }
+
+    if (engine.getValue(group, "loop_end_position") !== -1) {
+        Mixage.toggleLED(ON, group, "reloop");
+    } else {
+        Mixage.toggleLED(OFF, group, "reloop");
+    }
+};
+
+// resets the loop LEDs when a track is ejected
+Mixage.eject = function(group) {
+    if (engine.getValue(group, "play") === 0) {
+        if (Mixage.adjustLoop[group]) {
+            Mixage.stopLoopAdjust();
+        } else {
+            Mixage.blinkLED(Mixage.ledMap[group].reloop, group, 0);
+            Mixage.toggleLED(OFF, group, "loop");
+        }
+    }
+};
```

Comment:
Correct me if I'm wrong: the mapping does not provide eject, so this is called when `eject` is toggled in the GUI / via keyboard. Also it does not filter the value so it acts on press **and** release.

I suggest to merge this with the callback `Mixage.loopConnection.push(..)` (`track_loaded`) above.

Reply:
This was only called when `eject` was toggled in the gui, I merged the `Mixage.stopLoopAdjust();`  callback with the `track_loaded` engine connection